packages/service-utils/src/core/rateLimit/index.ts

```diff
@@ -1,49 +1,34 @@
 import type { CoreServiceConfig, TeamResponse } from "../api.js";
 import type { RateLimitResult } from "./types.js";
 
-const RATE_LIMIT_WINDOW_SECONDS = 10;
+const SLIDING_WINDOW_SECONDS = 10;
 
 // Redis interface compatible with ioredis (Node) and upstash (Cloudflare Workers).
 type IRedis = {
-  get: (key: string) => Promise<string | null>;
-  expire(key: string, seconds: number): Promise<number>;
   incrby(key: string, value: number): Promise<number>;
+  mget(keys: string[]): Promise<(string | null)[]>;
+  expire(key: string, seconds: number): Promise<number>;
 };
 
+/**
+ * Increments the request count for this team and returns whether the team has hit their rate limit.
+ * Uses a sliding 10 second window.
+ * @param args
+ * @returns
+ */
 export async function rateLimit(args: {
   team: TeamResponse;
   limitPerSecond: number;
   serviceConfig: CoreServiceConfig;
   redis: IRedis;
-  /**
-   * Sample requests to reduce load on Redis.
-   * This scales down the request count and the rate limit threshold.
-   * @default 1.0
-   */
-  sampleRate?: number;
   /**
    * The number of requests to increment by.
    * @default 1
    */
   increment?: number;
 }): Promise<RateLimitResult> {
-  const {
-    team,
-    limitPerSecond,
-    serviceConfig,
-    redis,
-    sampleRate = 1.0,
-    increment = 1,
-  } = args;
-
-  const shouldSampleRequest = Math.random() < sampleRate;
```

Comment:
Removed sampling, no service is using it (and if they need it its trivial to sample calls to this method).